stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -505,14 +508,23 @@ swift_conformsToProtocolImpl(const Metadata *const type,
       processSection(section);
   }
 
-  // Try the search again to look for the most specific cached conformance.
-  found = searchInConformanceCache(type, protocol);
+  // Find the most specific conformance that was scanned.
+  const WitnessTable *foundWitness = nullptr;
+  const Metadata *searchType = type;
+  while (!foundWitness && searchType) {
+    foundWitness = foundWitnesses.lookup(searchType);
 
-  // If it's not authoritative, then add an authoritative entry for this type.
-  if (!found.first)
-    C.cacheResult(type, protocol, found.second, snapshot.count());
+    // If there's no entry here, move up to the superclass (if any).
+    if (!foundWitness)
```

Comment:
I believe this test will fail to distinguish between "value not found" and "`NULL` found".

Reply:
Oh, t`C` is not the same as `foundWitnesses`. `foundWitnesses` is local. It's a local variable. It's a variable that's only used locally.